Non-specific debug output

Output from running `infrasetup` when there is a chisel compile-time error does not display any information about the error itself.

The logfile contained the same information that the stdout messages said.
As an example, I added the line

`var`

to the middle of `/sim/src/main/scala/endpoints/BlockDevWidget.scala` and then ran `firesim infrasetup`.

Output:
```
FireSim Manager. Docs: http://docs.fires.im
Running: infrasetup

Building FPGA software driver for FireSimNoNIC-FireSimRocketChipQuadCoreConfig-FireSimDDR3FRFCFSLLC4MBConfig
Fatal error.
Traceback (most recent call last):
  File "/home/centos/firesim/deploy/firesim", line 300, in <module>
    main(args)
  File "/home/centos/firesim/deploy/firesim", line 249, in main
    globals()[args.task](simconf)
  File "/home/centos/firesim/deploy/firesim", line 66, in infrasetup
    runtime_conf.infrasetup()
  File "/home/centos/firesim/deploy/runtools/runtime_config.py", line 240, in infrasetup
    self.firesim_topology_with_passes.infrasetup_passes(use_mock_instances_for_testing)
  File "/home/centos/firesim/deploy/runtools/firesim_topology_with_passes.py", line 293, in infrasetup_passes
    self.pass_build_required_drivers()
  File "/home/centos/firesim/deploy/runtools/firesim_topology_with_passes.py", line 276, in pass_build_required_drivers
    server.get_server_hardware_config().build_fpga_driver()
  File "/home/centos/firesim/deploy/runtools/runtime_config.py", line 110, in build_fpga_driver
    localcap = local("""make DESIGN={} TARGET_CONFIG={} PLATFORM_CONFIG={} f1""".format(design, target_config, platform_config), capture=True)
  File "/usr/lib64/python2.7/site-packages/fabric/operations.py", line 1239, in local
    error(message=msg, stdout=out, stderr=err)
  File "/usr/lib64/python2.7/site-packages/fabric/utils.py", line 358, in error
    return func(message)
  File "/usr/lib64/python2.7/site-packages/fabric/utils.py", line 62, in abort
    raise e
SystemExit: 1
The full log of this run is:
/home/centos/firesim/deploy/logs/2018-09-05--21-03-19-infrasetup-GD5CNBBYVG0WGN1O.log
```

And `cat /home/centos/firesim/deploy/logs/2018-09-05--21-03-19-infrasetup-GD5CNBBYVG0WGN1O.log` shows:
```
2018-09-05 21:03:19,468 [main        ] [INFO ]  FireSim Manager. Docs: http://docs.fires.im
Running: infrasetup

2018-09-05 21:03:19,468 [__init__    ] [DEBUG]  {'hwconf_dict': {'fireboom-singlecore-no-nic-ddr3-llc4mb': <runtools.runtime_config.RuntimeHWConfig instance at 0x7f2531a1ca28>,
                 'fireboom-singlecore-no-nic-lbp': <runtools.runtime_config.RuntimeHWConfig instance at 0x7f2531a1cb00>,
                 'firesim-quadcore-nic-ddr3-llc4mb': <runtools.runtime_config.RuntimeHWConfig instance at 0x7f2531a1cbd8>,
                 'firesim-quadcore-nic-lbp': <runtools.runtime_config.RuntimeHWConfig instance at 0x7f2531a1cb90>,
                 'firesim-quadcore-no-nic-ddr3-llc4mb': <runtools.runtime_config.RuntimeHWConfig instance at 0x7f2531a1cab8>,
                 'firesim-quadcore-no-nic-lbp': <runtools.runtime_config.RuntimeHWConfig instance at 0x7f2531a1ca70>,
                 'firesim-singlecore-nic-lbp': <runtools.runtime_config.RuntimeHWConfig instance at 0x7f2531a1cc20>,
                 'firesim-singlecore-no-nic-lbp': <runtools.runtime_config.RuntimeHWConfig instance at 0x7f2531a1cb48>}}
2018-09-05 21:03:19,470 [__init__    ] [DEBUG]  {'defaulthwconfig': 'firesim-quadcore-no-nic-ddr3-llc4mb',
 'f1_16xlarges_requested': 0,
 'f1_2xlarges_requested': 1,
 'linklatency': 6405,
 'm4_16xlarges_requested': 0,
 'netbandwidth': 200,
 'no_net_num_nodes': 1,
 'run_instance_market': 'ondemand',
 'runfarmtag': 'mainrunfarm',
 'spot_interruption_behavior': 'terminate',
 'spot_max_price': 'ondemand',
 'switchinglatency': 10,
 'terminateoncompletion': False,
 'topology': 'no_net_config',
 'workload_name': 'linux-uniform.json'}
2018-09-05 21:03:19,470 [get_deploytr] [DEBUG]  Setting deploytriplet by querying the AGFI's description.
2018-09-05 21:03:19,470 [get_afi_for_] [DEBUG]  agfi-00f8511c144ca7774
2018-09-05 21:03:19,470 [get_afi_for_] [DEBUG]  None
2018-09-05 21:03:20,132 [get_afi_for_] [DEBUG]  {u'FpgaImages': [{u'UpdateTime': datetime.datetime(2018, 6, 14, 3, 8, 15, tzinfo=tzlocal()), u'Name': 'firesim-quadcore-no-nic-ddr3-llc4mb', u'Tags': [], u'PciId': {u'SubsystemVendorId': '0xfedd', u'VendorId': '0x1d0f', u'DeviceId': '0xf000', u'SubsystemId': '0x1d51'}, u'FpgaImageGlobalId': 'agfi-00f8511c144ca7774', u'Public': True, u'State': {u'Code': 'available'}, u'ShellVersion': '0x071417d3', u'OwnerId': '552479941143', u'FpgaImageId': 'afi-0f2841a94d42f0b4e', u'CreateTime': datetime.datetime(2018, 6, 14, 2, 35, 24, tzinfo=tzlocal()), u'Description': 'firesim-buildtriplet:FireSimNoNIC-FireSimRocketChipQuadCoreConfig-FireSimDDR3FRFCFSLLC4MBConfig,firesim-deploytriplet:FireSimNoNIC-FireSimRocketChipQuadCoreConfig-FireSimDDR3FRFCFSLLC4MBConfig,firesim-commit:c33fa554c8c155ac4442be658ab562a1163b7d5f-dirty'}], 'ResponseMetadata': {'RetryAttempts': 0, 'HTTPStatusCode': 200, 'RequestId': 'a266c452-f30e-4939-9757-6aca88c9327a', 'HTTPHeaders': {'transfer-encoding': 'chunked', 'vary': 'Accept-Encoding', 'server': 'AmazonEC2', 'content-type': 'text/xml;charset=UTF-8', 'date': 'Wed, 05 Sep 2018 21:03:19 GMT'}}}
2018-09-05 21:03:20,909 [build_fpga_d] [INFO ]  Building FPGA software driver for FireSimNoNIC-FireSimRocketChipQuadCoreConfig-FireSimDDR3FRFCFSLLC4MBConfig
2018-09-05 21:03:20,909 [flush       ] [DEBUG]  [localhost] local: make DESIGN=FireSimNoNIC TARGET_CONFIG=FireSimRocketChipQuadCoreConfig PLATFORM_CONFIG=FireSimDDR3FRFCFSLLC4MBConfig f1
2018-09-05 21:03:46,716 [flush       ] [DEBUG]  Fatal error: local() encountered an error (return code 2) while executing 'make DESIGN=FireSimNoNIC TARGET_CONFIG=FireSimRocketChipQuadCoreConfig PLATFORM_CONFIG=FireSimDDR3FRFCFSLLC4MBConfig f1'
2018-09-05 21:03:46,717 [flush       ] [DEBUG]  Aborting.
2018-09-05 21:03:46,717 [<module>    ] [ERROR]  Fatal error.
Traceback (most recent call last):
  File "/home/centos/firesim/deploy/firesim", line 300, in <module>
    main(args)
  File "/home/centos/firesim/deploy/firesim", line 249, in main
    globals()[args.task](simconf)
  File "/home/centos/firesim/deploy/firesim", line 66, in infrasetup
    runtime_conf.infrasetup()
  File "/home/centos/firesim/deploy/runtools/runtime_config.py", line 240, in infrasetup
    self.firesim_topology_with_passes.infrasetup_passes(use_mock_instances_for_testing)
  File "/home/centos/firesim/deploy/runtools/firesim_topology_with_passes.py", line 293, in infrasetup_passes
    self.pass_build_required_drivers()
  File "/home/centos/firesim/deploy/runtools/firesim_topology_with_passes.py", line 276, in pass_build_required_drivers
    server.get_server_hardware_config().build_fpga_driver()
  File "/home/centos/firesim/deploy/runtools/runtime_config.py", line 110, in build_fpga_driver
    localcap = local("""make DESIGN={} TARGET_CONFIG={} PLATFORM_CONFIG={} f1""".format(design, target_config, platform_config), capture=True)
  File "/usr/lib64/python2.7/site-packages/fabric/operations.py", line 1239, in local
    error(message=msg, stdout=out, stderr=err)
  File "/usr/lib64/python2.7/site-packages/fabric/utils.py", line 358, in error
    return func(message)
  File "/usr/lib64/python2.7/site-packages/fabric/utils.py", line 62, in abort
    raise e
SystemExit: 1
2018-09-05 21:03:46,722 [<module>    ] [INFO ]  The full log of this run is:
/home/centos/firesim/deploy/logs/2018-09-05--21-03-19-infrasetup-GD5CNBBYVG0WGN1O.log
```

So the main error messages we can see here are `Fatal error.` and `SystemExit: 1`. I think @sagark had mentioned that he knew what was going wrong here and how to fix it.

--- a/deploy/firesim
+++ b/deploy/firesim
@@ -105,12 +105,12 @@ def buildafi(globalbuildconf):
 
     signal.signal(signal.SIGINT, terminate_instances_handler)
 
+    for buildconf in globalbuildconf.get_builds_list():
+        execute(replace_rtl, globalbuildconf, buildconf, hosts=['localhost'])
+
     # local items (replace_rtl) need to be called in a loop, for each config
     # remote items will map themselves
     globalbuildconf.launch_build_instances()
-
-    for buildconf in globalbuildconf.get_builds_list():
-        execute(replace_rtl, globalbuildconf, buildconf, hosts=['localhost'])
 
     # confirm that build instances have finished booting
     globalbuildconf.wait_build_instances()

--- a/deploy/runtools/runtime_config.py
+++ b/deploy/runtools/runtime_config.py
@@ -171,9 +171,16 @@ class RuntimeHWConfig:
         platform_config = triplet_pieces[2]
         rootLogger.info("Building FPGA software driver for " + str(self.get_deploytriplet_for_config()))
         with prefix('cd ../'), prefix('source sourceme-f1-manager.sh'), prefix('cd sim/'), StreamLogger('stdout'), StreamLogger('stderr'):
-            localcap = local("""make DESIGN={} TARGET_CONFIG={} PLATFORM_CONFIG={} f1""".format(design, target_config, platform_config), capture=True)
+            localcap = None
+            with settings(warn_only=True):
+                driverbuildcommand = """make DESIGN={} TARGET_CONFIG={} PLATFORM_CONFIG={} f1""".format(design, target_config, platform_config)
+                localcap = local(driverbuildcommand, capture=True)
             rootLogger.debug("[localhost] " + str(localcap))
             rootLogger.debug("[localhost] " + str(localcap.stderr))
+            if localcap.failed:
+                rootLogger.info("FPGA software driver build failed. Exiting. See log for details.")
+                rootLogger.info("""You can also re-run '{}' in the 'firesim/sim' directory to debug this error.""".format(driverbuildcommand))
+                exit(1)
 
         self.driver_built = True
 

